### DISCUSSION TITLE
Programmatically export Excel "Scarica dettaglio righe"

### DISCUSSION THREAD

**Discussion by marioquark** [fattureincloud/api#479] [Q&A] [answered]
Hi, I am trying to understand if there is way to programmatically download **Excel reports via API**.
I need the 4 reports generated using the button "Scarica dettaglio righe" in the UI in those pages: Fatture, Ricevute, Acquisti and Note di credito.
I cannot see anything in API Reference, since APIs are intended to extract JSONs and not Xls sheets.
Thanks,
Mario

**Comment by valmoz** [comment]
Hello,
no, our APIs only provide access to the data using JSON format.

You can though decide to turn the data returned by our APIs into other formats, such as XLSX, using libraries similar to [Apache POI](https://poi.apache.org/) (it depends on the language you're using to create your integration, or if you're using a No-Code software like [Zapier](https://developers.fattureincloud.it/docs/quickstarts/zapier-quickstart)).
You could also try to download CSV or JSON files, and to convert them directly using Excel.

Thanks
Mauro

**Comment by marioquark** [comment]
Thanks Valmoz,
the CSV download could be super ok as solution. 
Is there any API to have those reports downloaded as CSV?
Thanks

**Reply by valmoz** [reply, marked as answer]
> no, our APIs only provide access to the data using JSON format.

As stated before, we only support JSON format, but turning a JSON into CSV should be much easier than doing the same action with XLSX. Some JSON parsing libraries natively support CSV too.

Let us know if you need additional info.

Thanks
Mauro
